### ISSUE TITLE
Fresh project setup issue.

### ISSUE THREAD

**Issue by Mahadeb369** [apptension/saas-boilerplate#340]
### Describe the bug

Running `make up` on a fresh repository throws an error because **workers** and **backend** images are not building properly.

### Steps to reproduce

1. Clone the repo, `git clone -b 2.0.1 https://github.com/apptension/saas-boilerplate.git`
2. Install dependencies, `pnpm install`
3. Start the backend, `make up`

### System Info

```shell
System:
    OS: Linux 5.4 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (4) x64 AMD EPYC Processor
    Memory: 20.35 GB / 28.70 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.8.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
    pnpm: 8.6.9 - ~/.nvm/versions/node/v18.16.0/bin/pnpm
  npmPackages:
    @apollo/client: ^3.7.9 => 3.7.9 
    @apollo/rover: ^0.12.2 => 0.12.2 
    @babel/preset-react: ^7.14.5 => 7.18.6 
    @graphql-typed-document-node/core: ^3.2.0 => 3.2.0 
    @nx/devkit: 16.3.2 => 16.3.2 
    @nx/eslint-plugin: 16.3.2 => 16.3.2 
    babel-jest: 29.4.3 => 29.4.3 
    babel-plugin-relay: ^14.1.0 => 14.1.0 
    babel-plugin-styled-components: 1.10.7 => 1.10.7 
    env-cmd: ^10.1.0 => 10.1.0 
    eslint: ^8.34.0 => 8.34.0 
    eslint-config-prettier: ^8.5.0 => 8.6.0 
    eslint-import-resolver-typescript: ^3.5.3 => 3.5.3 
    eslint-plugin-cypress: ^2.10.3 => 2.12.1 
    eslint-plugin-import: 2.27.5 => 2.27.5 
    eslint-plugin-jsx-a11y: 6.7.1 => 6.7.1 
    eslint-plugin-react: 7.32.2 => 7.32.2 
    eslint-plugin-react-hooks: 4.6.0 => 4.6.0 
    graphql: ^16.6.0 => 16.6.0 
    husky: ^8.0.0 => 8.0.3 
    jest: 29.4.3 => 29.4.3 
    jest-environment-jsdom: 29.4.3 => 29.4.3 
    jsonc-eslint-parser: ^2.1.0 => 2.1.0 
    lint-staged: ^10.5.4 => 10.5.4 
    nx: 16.3.2 => 16.3.2 
    nx-cloud: 16.0.5 => 16.0.5 
    prettier: ^2.6.2 => 2.8.3 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    react-is: 18.2.0 => 18.2.0 
    react-refresh: ^0.10.0 => 0.10.0 
    react-router-dom: 6.11.2 => 6.11.2 
    styled-components: 5.3.6 => 5.3.6 
    tailwindcss: ^3.3.2 => 3.3.2 
    tailwindcss-animate: ^1.0.5 => 1.0.5 
    ts-jest: 29.1.0 => 29.1.0 
    ts-node: 10.9.1 => 10.9.1 
    tsconfig-paths: ^4.1.2 => 4.1.2 
    tslib: ^2.3.0 => 2.4.1 
    typescript: 5.0.4 => 5.0.4 
    url-loader: ^4.1.1 => 4.1.1
```

### Logs

```shell
core:  >  NX   Successfully ran target docker-create-volumes for project core
core:
core:  >  NX   Successfully ran target setup for 5 projects and 1 task they depend on
core:    Nx read the output from the cache instead of running the command for 5 out of 6 tasks.
core:  >  NX   Running target compose-build-image for 2 projects and 2 tasks they depend on:
core:     - backend
core:     - workers
core:
core: > nx run backend:setup  [existing outputs match the cache, left as is]
core: > nx run workers:setup  [existing outputs match the cache, left as is]
core: > nx run workers:compose-build-image
core: > nx run core:docker-login
core:
core:  >  NX   Successfully ran target docker-login for project core
core: The Compose file './docker-compose.yml' is invalid because:
core: services.workers.depends_on contains an invalid type, it should be an array
core: services.backend.depends_on contains an invalid type, it should be an array
core: Warning: run-commands command "docker-compose build workers" exited with non-zero status code
core: > nx run backend:compose-build-image
core: > nx run core:docker-login
core:
core:  >  NX   Successfully ran target docker-login for project core
core: The Compose file './docker-compose.yml' is invalid because:
core: services.workers.depends_on contains an invalid type, it should be an array
core: services.backend.depends_on contains an invalid type, it should be an array
core: Warning: run-commands command "docker-compose build backend" exited with non-zero status code
core:
core:  >  NX   Running target compose-build-image for 2 projects and 2 tasks they depend on failed
core:    Failed tasks:
core:
core:    - workers:compose-build-image
core:    - backend:compose-build-image
core: Warning: run-commands command "nx run-many --target=compose-build-image --projects=backend,workers" exited with non-zero status code

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target docker-compose:up for project core and 2 tasks it depends on failed

   Failed tasks:

   - core:docker-compose:up

   Hint: run the command with --verbose for more details.

make: *** [Makefile.base.mk:37: up] Error 1
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/apptension/saas-boilerplate/blob/master/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/apptension/saas-boilerplate/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://docs.demo.saas.apptoku.com/).
- [X] Check that there isn't [already an issue](https://github.com/apptension/saas-boilerplate/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/apptension/saas-boilerplate/discussions) or join our [Discord Chat Server](https://discord.apptension.com).

**Comment by pziemkowski** [comment]
Would you be able to add full logs? I can see 
```
core:    Failed tasks:
core:
core:    - workers:compose-build-image
core:    - backend:compose-build-image
```

but without logs from those particular tasks it's hard to say what happened

**Comment by pziemkowski** [comment]
Actually, it looks like someone introduced  Docker compose v2 syntax into our docker-compose.yaml file

**Comment by Mahadeb369** [comment]
I'm also not able to see logs of these images. 
When I'm running the command `nx run workers:compose-build-image --verbose`  and

**Comment by Mahadeb369** [comment]
I'm also not able to see logs of these images.
When I'm running these commands, `nx run workers:compose-build-image --verbose` and `nx run backend:compose-build-image --verbose`, these are not giving me much information.

Logs of `nx run workers:compose-build-image --verbose`
```shell
root@e2e-45-12 /e/m/saas-app ((2.0.1)) [2]# nx run workers:compose-build-image --verbose

 >  NX   Running global Nx CLI with PNPM may have issues.

   Prefer to use "pnpm" (https://pnpm.io/cli/exec) to execute commands in this workspace.
    TIP  create a shortcut such as: alias pnx="pnpm nx --"
   


 >  NX   Running target compose-build-image for project workers and 1 task it depends on:

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> nx run workers:setup  [existing outputs match the cache, left as is]


> nx run workers:compose-build-image

> nx run core:docker-login
 
 >  NX   Successfully ran target docker-login for project core
The Compose file './docker-compose.yml' is invalid because:
services.workers.depends_on contains an invalid type, it should be an array
services.backend.depends_on contains an invalid type, it should be an array
Warning: run-commands command "docker-compose build workers" exited with non-zero status code

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target compose-build-image for project workers and 1 task it depends on failed

   Failed tasks:
   
   - workers:compose-build-image
   
   Hint: run the command with --verbose for more details.
   
```

Logs of `nx run backend:compose-build-image --verbose`
```shell
root@e2e-45-12 /e/m/vidgen-webapp ((2.0.1)) [1]# nx run backend:compose-build-image --verbose

 >  NX   Running global Nx CLI with PNPM may have issues.

   Prefer to use "pnpm" (https://pnpm.io/cli/exec) to execute commands in this workspace.
    TIP  create a shortcut such as: alias pnx="pnpm nx --"
   


 >  NX   Running target compose-build-image for project backend and 1 task it depends on:

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> nx run backend:setup  [existing outputs match the cache, left as is]


> nx run backend:compose-build-image

> nx run core:docker-login
 
 >  NX   Successfully ran target docker-login for project core
The Compose file './docker-compose.yml' is invalid because:
services.workers.depends_on contains an invalid type, it should be an array
services.backend.depends_on contains an invalid type, it should be an array
Warning: run-commands command "docker-compose build backend" exited with non-zero status code

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target compose-build-image for project backend and 1 task it depends on failed

   Failed tasks:
   
   - backend:compose-build-image
   
   Hint: run the command with --verbose for more details.
   
```

**Comment by pziemkowski** [comment]
Try upgrading your docker-compose to version 2. Seems like you're using old version of docker-compose

**Comment by pziemkowski** [comment]
I managed to reproduce it locally. Seems like some dependency of a dependency was updated and broke a lot of projects.

**Comment by pziemkowski** [comment]
As a quick fix adding `"pyyaml>=6.0.0"` to `pyproject.toml` in `packages/backend` seems to work.
